### DISCUSSION TITLE
Bonsai VideoWriter saving videos at variable rates

### DISCUSSION THREAD

**Discussion by lancygnos** [bonsai-rx/bonsai#1737] [Q&A]
Hi!

I’m trying to record mice behaviors but have been encountering an issue where the avi files are saving at variable rates. I tried a simpler workflow as attached and the problem still persists.

Specifically, when I set the Timer node to 10 sec and the video format to 30fps (all of the available formats in Bonsai are 30 fps), the saved avi file only lasts for 9 seconds. The interval between timestamps also appears to fluctuate because the video noticeably speeds up towards the end. Some solutions I’ve tried:

Altering the frame rate: when I change the frame rate to 24 fps, the video becomes 11 seconds.

Switching to another camera: interestingly the output becomes 8 seconds for the same workflow and frame rate. 

Changing the computer: the same camera and workflow saves the video at a different duration. 

I also tried changing the Exposure to Manual for VideoCaptureDevice as suggested in https://groups.google.com/g/bonsai-users/c/y0kmWNRvS84, but when I did that, the video was so sped up that the duration was barely 1 second. 

I’m not so sure about the root cause of the problem, so I would greatly appreciate any assistance with this!

![image](https://github.com/bonsai-rx/bonsai/assets/167103234/025d51f0-567e-4dda-87f9-21af1311e421)

**Comment by bruno-f-cruz** [comment]
Sounds like the camera is not acquiring at a stable framerate. Are you sure that the autoexposure is being correctly turned off? Does the frame-rate vary with the amount of light in the environment? This discussion has been had several times in the past, it is probably a good idea to read the other discussions (e.g. https://github.com/orgs/bonsai-rx/discussions/1112)
